### PR TITLE
Add com.notesnook.Notesnook

### PR DESCRIPTION
**Pull request by thecodrr**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.streetwriters.notesnook.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+base: org.electronjs.Electron2.BaseApp
+base-version: "22.08"
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node16
+command: start-notesnook
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=system-bus
```
Comment (hfiguiere) [inline com.streetwriters.notesnook.yml:14]: You need to figure which D-Bus name you need. This is not allowed.

File: com.streetwriters.notesnook.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      - install -Dm644 ./apps/web/desktop/assets/icons/128x128.png  /app/share/icons/hicolor/128x128/apps/$FLATPAK_ID.png
+      - install -Dm644 ./apps/web/desktop/assets/icons/256x256.png  /app/share/icons/hicolor/256x256/apps/$FLATPAK_ID.png
+      - install -Dm644 ./apps/web/desktop/assets/icons/512x512.png  /app/share/icons/hicolor/512x512/apps/$FLATPAK_ID.png
+      - install -D com.streetwriters.notesnook.desktop -t  /app/share/applications/
+      - install -D com.streetwriters.notesnook.metainfo.xml -t  /app/share/metainfo/
+    sources:
+      - generated-sources.json
+      - type: git
+        url: https://github.com/streetwriters/notesnook.git
+        branch: master
```
Comment (hfiguiere) [inline com.streetwriters.notesnook.yml:62]:
```suggestion
```
NEVER git branches.

File: com.streetwriters.notesnook.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+app-id: com.streetwriters.notesnook
```
Comment (hfiguiere) [inline com.streetwriters.notesnook.yml:1]: `streetwriters.com` seems to be unrelated. Also the project list `https://notesnook.com/`, so I don't think the app-id meet the requirements.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build com.streetwriters.notesnook

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/com.streetwriters.notesnook.appdata.xml: FAILED:
• style-invalid         : Too many <p> tags for a good description [27/10]
Validation of files failed
```

File: com.streetwriters.notesnook.yml (hunk 2 above)
Comment (thecodrr) [reply com.streetwriters.notesnook.yml:62]: If you see the line below it, it includes a specific git commit hash. From the docs:

> commit (string)
> The commit to use from the git repository. If branch is also specified, then it is verified that the branch/tag is at this specific commit. This is a readable way to document that you're using a particular tag, but verify that it does not change.

File: com.streetwriters.notesnook.yml (hunk 1 above)
Comment (thecodrr) [reply com.streetwriters.notesnook.yml]: Fixed

File: com.streetwriters.notesnook.yml (hunk 3 above)
Comment (thecodrr) [reply com.streetwriters.notesnook.yml:1]: Notesnook runs under the parent company Streetwriters that owns the domain streetwriters.co

Can `co.streetwriters.notesnook` be used?

Comment (hfiguiere) [reply com.streetwriters.notesnook.yml:1]: `co` is not `com` mind you.

Comment (hfiguiere) [reply com.streetwriters.notesnook.yml:1]: But given the app data points to `    <url type="homepage">https://notesnook.com/</url>`

`com.notesnook.notesnook` seems to be the appropriate app-id in my books.

File: com.streetwriters.notesnook.yml (hunk 2 above)
Comment (thecodrr) [reply com.streetwriters.notesnook.yml]: Done

File: com.streetwriters.notesnook.yml (hunk 3 above)
Comment (thecodrr) [reply com.streetwriters.notesnook.yml]: Done.

**Comment by hfiguiere** [comment]
have the desktop and appstream file been submitted upstream?

A desktop app without a .desktop file is a bug.

**Comment by hfiguiere** [comment]
bot, build com.notesnook.Notesnook

**Comment by thecodrr** [comment]
The upstream project uses Electron Builder which automatically generates the .desktop file. Flatpak is an exception so it is handled separately. Do note that I am the maintainer of the upstream project as well.

**Comment by thecodrr** [comment]
Thank you!
